

|---+----------+--------+----------------------------+---------------+---------------+--------+--------+-------|
|   | Gene     | Length | Region                     | Variants Type | Variant Count | source |      N |     m |
|---+----------+--------+----------------------------+---------------+---------------+--------+--------+-------|
|   |          |        |                            |               |               |        |        |       |
|---+----------+--------+----------------------------+---------------+---------------+--------+--------+-------|
| 6 | HLA-B    | 13,193 | chr6:31,353,875-31,367,067 | all           |          4483 | all    |        |       |
|   |          |        |                            |               |            74 | 1000G  |  13332 |   224 |
|   |          |        |                            |               |          1188 | TOPMed |  15090 |  3453 |
|   |          |        |                            |               |           663 | gnomAD |  14436 |  2044 |
|   |          |        |                            | SNP           |          3752 | all    |        |       |
|   |          |        |                            |               |            53 | 1000G  |  13247 |   161 |
|   |          |        |                            |               |          1075 | TOPMed |  14355 |  3170 |
|   |          |        |                            |               |           567 | gnomAD |  14355 |  3170 |
|---+----------+--------+----------------------------+---------------+---------------+--------+--------+-------|
|   | HLA-DRB1 | 11,074 | chr6:32,578,775-32,589,848 | all           |         6,477 | all    |        |       |
|   |          |        |                            |               |            99 | 1000G  | 11,278 |   307 |
|   |          |        |                            |               |         2,196 | TOPMed | 15,470 | 6,358 |
|   |          |        |                            |               |           368 | gnomAD | 12,004 | 1,091 |
|   |          |        |                            | SNP           |          4979 | all    |        |       |
|   |          |        |                            |               |            70 | 1000G  | 11,074 |     1 |
|   |          |        |                            |               |         1,939 | TOPMed | 13,451 | 5,749 |
|   |          |        |                            |               |           285 | gnomAD | 11,425 |   870 |
|---+----------+--------+----------------------------+---------------+---------------+--------+--------+-------|
| 7 | ACTB     | 36,751 | chr7:5,527,152-5,563,902   | all           |         12683 | all    |        |       |
|   |          |        |                            |               |           212 | 1000G  |  37019 |   644 |
|   |          |        |                            |               |          5275 | TOPMed |  46944 | 15413 |
|   |          |        |                            |               |          1210 | gnomAD |  39006 |  3621 |
|   |          |        |                            | SNP           |         11399 | all    |        |       |
|   |          |        |                            |               |           208 | 1000G  |  36967 |   630 |
|   |          |        |                            |               |          4960 | TOPMed |  41947 | 13891 |
|   |          |        |                            |               |          1044 | gnomAD |  37876 |  3146 |


* Download links

Human genome data used *GRCh38.p13*.

Downloaded from ensembl

Using ~aria2~ on the download links in the file download-links
For example ~aria2c --continue -i download-links.txt~


** Chr 6

ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.6.fa.gz
ftp://ftp.ensembl.org/pub/release-109/variation/vcf/homo_sapiens/homo_sapiens-chr6.vcf.gz

** Chr 7

ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.7.fa.gz
ftp://ftp.ensembl.org/pub/release-109/variation/vcf/homo_sapiens/homo_sapiens-chr7.vcf.gz


* Regions

** ACTB

bedfile
#+BEGIN_SRC bash
7	5527152	5563902
#+END_SRC

bedtools to create sub fa
#+BEGIN_SRC bash
REF=Homo_sapiens.GRCh38.dna.chromosome.7.fa; GENE=actb; \
bedtools getfasta -fi $REF -bed $GENE.bed -fo chr7.$GENE.fa
#+END_SRC


* Commands
Index the VCF
#+BEGIN_SRC bash
 gatk IndexFeatureFile \
     -F homo_sapiens-chr7.vcf
#+END_SRC


Count variants
#+BEGIN_SRC bash
REGION="7:5527152-5563902"; gatk CountVariants \
     -R Homo_sapiens.GRCh38.dna.chromosome.7.fa \
     -V homo_sapiens-chr7.vcf \
     -L $REGION
#+END_SRC



Select all variants
#+BEGIN_SRC bash
REF=Homo_sapiens.GRCh38.dna.chromosome.7.fa; REGION="7:5527152-5563902"; gatk SelectVariants \
     -R $REF \
     -V homo_sapiens-chr7.vcf \
     -L $REGION \
     -O actb.indels.vcf
#+END_SRC

#+BEGIN_SRC bash
REF=Homo_sapiens.GRCh38.dna.chromosome.7.fa; REGION="7:5527152-5563902";  VARIANT=SNP; OUT=actb.snp.vcf; gatk SelectVariants \
  -R $REF \
  -V homo_sapiens-chr7.vcf \
  --select-type-to-include $VARIANT \
  -L $REGION \
  -O $OUT && \
gatk CountVariants \
  -R $REF \
  -V $OUT \
  -L $REGION
#+END_SRC


from DB
no variant
#+BEGIN_SRC bash
REF=Homo_sapiens.GRCh38.dna.chromosome.7.fa; REGION="7:5527152-5563902"; OUT="actb.topmed.indels.vcf"; \
gatk SelectVariants \
  -R $REF \
  -V homo_sapiens-chr7.vcf \
  -select "E_TOPMed" \
  -L $REGION \
  -O x && \
gatk SelectVariants \
  -R $REF \
  -V x \
  -invert-select -select "E_gnomAD" -select "E_1000G" \
  -L $REGION \
  -O $OUT && \
gatk CountVariants \
  -R $REF \
  -V $OUT \
  -L $REGION
#+END_SRC


with variant
#+BEGIN_SRC bash
REF=Homo_sapiens.GRCh38.dna.chromosome.7.fa; REGION="7:5527152-5563902"; VARIANT=SNP; OUT="actb.1000G.$VARIANT.vcf"; \
gatk SelectVariants \
  -R $REF \
  -V homo_sapiens-chr7.vcf \
  --select-type-to-include $VARIANT \
  -select "E_1000G" \
  -L $REGION \
  -O x && \
gatk SelectVariants \
  -R $REF \
  -V x \
  --select-type-to-include $VARIANT \
  -invert-select -select "E_gnomAD" -select "E_TOPMed" \
  -L $REGION \
  -O $OUT && \
gatk CountVariants \
  -R $REF \
  -V $OUT \
  -L $REGION
#+END_SRC

Gen ~.eds~ with aedso
#+BEGIN_SRC
REF=chr7.$GENE.fa; GENE=actb; SOURCE=1000G; TYPE=indels; START=5527152; STOP=5563902; \
aedso -v -s $START -e $STOP $REF $GENE.$SOURCE.$TYPE.vcf > $GENE.$SOURCE.$TYPE.eds
#+END_SRC

